### ISSUE TITLE
Helm pre-install hooks prevent resource creation in app-of-apps deployments.

### ISSUE THREAD

**Issue by ryanhockstad** [argoproj/argo-cd#16778]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

argocd: v2.7.14+a40c95a
  BuildDate: 2023-09-07T17:39:29Z
  GitCommit: a40c95a6383baa0d9f670586a5c295021f59337c
  GitTreeState: clean
  GoVersion: go1.19.12
  Compiler: gc
  Platform: linux/arm64

**Describe the bug**
When using an app of apps pattern, presync hooks in the sub-applications prevent their corresponding  resources from being deployed.

**To Reproduce**

The following steps reproduce this issue:

* A helm chart that contains templates for other helm charts to be deployed as Argocd Applications is itself deployed as an ArgoCD Application with autosync disabled. (App of apps pattern)
* The sub-applications are configured to be deployed in sync waves, and some have templates with pre-sync annotations.
* Autosync is enabled for the sub-applications, and the outer-application is synced.
* After waiting for the applications to spin up, they appear to be synced, but all resources that are annotated with pre-sync hooks do not exist

**Expected behavior**

I expect the sub-applications should spin up such that resources annotated with pre-sync hooks are the first to be deployed instead of not being deployed at all.

**Screenshots**

**Version**

```shell
argocd: v2.7.14+a40c95a
  BuildDate: 2023-09-07T17:39:29Z
  GitCommit: a40c95a6383baa0d9f670586a5c295021f59337c
  GitTreeState: clean
  GoVersion: go1.19.12
  Compiler: gc
  Platform: linux/arm64
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by cardoe** [comment]
Agreed here. The post-install,post-upgrade Helm hook should really map to the Sync ArgoCD hook. In Helm the post-install,post-upgrade happens after all the resources have been applied but not before they're up and healthy. This maps to the ArgoCD Sync hook instead of the ArgoCD PostSync hook which happens after the resources are applied and healthy. I believe #17604 is related to this.

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and let us know if the issue is still present, please?
